math/cos_test.go

```diff
@@ -1,11 +1,12 @@
-package math
+package math_test
 
 import (
+	. "github.com/TheAlgorithms/Go/math"
```

Comment:
This is a bad practice. Just use the math package instead of importing everything in the `_test` space - call it `algmath` instead or something. Better yet change the standard library import name to `stdmath "math"` instead. Importing without a namespace is a bad practice in general whether its Go or some other language.

Reply:
> This is a bad practice. Just use the math package instead of importing everything in the `_test` space - call it `algmath` instead or something. Better yet change the standard library import name to `stdmath "math"` instead. Importing without a namespace is a bad practice in general whether its Go or some other language.

Good! poke me right?! I do not understand

Reply:
```suggestion
	algmath "github.com/TheAlgorithms/Go/math"
```

and when calling it in this package, you use `algmath` because its confined to the package scoped `algmath`
 like so

```Go
algmath.Cos(...)
```

The concept of namespacing was added to C++ in 1995 and the idea for this was to prevent name collisions and grouping similar logical code into blocks (This `std::vector` in the C++ standard library - `std` is the namespace for standard library utilities). The idea is the same as packages in Go or modules in Python. The reason is that now, if you were to define a function `Sin` or `Cos` in the `math_test` package then you would have errors. I also understand that this is a small package and thus you are unlikely to come across these issues but in big packages it is actually an issue (at least in  code bases I've worked on).

For more information, I prefer a lot – *not all* – of the Go styling conventions from [this](https://github.com/uber-go/guide/blob/master/style.md#import-aliasing)

Reply:
Yes, great, now I understand that you meant the namespace, this is very correct. Corrected it